protocol/directory.go

```diff
@@ -54,56 +54,86 @@ func (d *ConiksDirectory) LatestSTR() *merkletree.SignedTreeRoot {
 	return d.pad.LatestSTR()
 }
 
-func (d *ConiksDirectory) Register(uname string, key []byte) (
-	*merkletree.AuthenticationPath, *merkletree.TemporaryBinding, ErrorCode) {
+func (d *ConiksDirectory) Register(req *RegistrationRequest) (
+	*RegistrationResponse, ErrorCode) {
 	// check whether the name already exists
 	// in the directory before we register
-	ap, err := d.pad.Lookup(uname)
+	ap, err := d.pad.Lookup(req.Username)
 	if err != nil {
-		return nil, nil, ErrorInternalServer
+		return nil, ErrorDirectory
 	}
 	if bytes.Equal(ap.LookupIndex, ap.Leaf.Index()) {
-		return ap, nil, ErrorNameExisted
+		return &RegistrationResponse{
```

Comment:
I'm not so sure. Eventually, this's just a constructor. A setter function doesn't change the code so much, I think.